src/system.c

```diff
@@ -641,7 +646,7 @@ int execve( char *name, char **argv, char **env )
 void _exit( int rc )
 {
     /* Default stub just causes a divide by 0 exception.  */
-    int x = rc / INT_MAX;
+    int x = rc / INT_MAX; // TODO: what if rc == INT_MAX
```

Comment:
Uhm I'm not sure we ever get here. Surely a division by zero is not the correct thing to do, as it would cause an exception screen to show up. I think it would be better to just remove this code here and loop forever like we do on `abort()`.

There has been some discussion in the emulation community to standardize a way for homebrew code to declare the "end" of a program, so that the emulator could in theory exit; this would be useful when using emulators to run testsuites for instance, even more so in headless mode. But we haven't reached a consensus.

Reply:
isn't `_exit`, the "implementation" of abort?
https://sourceware.org/newlib/libc.html#Stubs
https://sourceware.org/git/?p=newlib-cygwin.git;a=blob;f=newlib/libc/stdlib/abort.c;hb=main#l59

(my initial point with "what if rc == INT_MAX" is it wouldn't cause a `/0` but it seems it's overshadowed by the above :thinking: )

Reply:
lol here is the assembly of `_exit`:
```
Disassembly of section .text._exit:

00000000 <_exit>:
   0:   1000ffff        b       0 <_exit>
   4:   00000000        nop
        ...
```

(
```
cd $N64_INST/mips64-elf/lib
$N64_INST/bin/mips64-elf-objdump libdragonsys.a -d -j.text._exit
```
)

big brain gcc

Reply:
Ok so I'd remove the division and just keep the infinite loop